File: 1561-maximum-number-of-coins-you-can-get/1561-maximum-number-of-coins-you-can-get.java
class Solution {
    public int maxCoins(int[] piles) {
        Arrays.sort(piles);
        int ans = 0;
        for(int i=piles.length/3; i<piles.length; i+=2) ans += piles[i];
        return ans;
    }
}


// class Solution {
//     public int maxCoins(int[] piles) {
//         Arrays.sort(piles);
//         int i=0;
//         int j=piles.length-2;
//         int ans = 0;
//         while(i<=j){
//             ans += piles[j];
//             ++i;
//             j-=2;
//         }
//         return ans;
//     }
// }

File: 1561-maximum-number-of-coins-you-can-get/1561-maximum-number-of-coins-you-can-get.py
class Solution:
    def maxCoins(self, piles: List[int]) -> int:
        piles.sort()
        i, j, ans = 0, len(piles)-2, 0
        while i<j:
            ans += piles[j]
            i+=1
            j-=2
        return ans